libminifi/src/RemoteProcessorGroupPort.cpp

```diff
@@ -167,17 +167,12 @@ void RemoteProcessorGroupPort::onSchedule(const std::shared_ptr<core::ProcessCon
     }
   }
   {
-    uint64_t idleTimeoutVal = 15000;
-    std::string idleTimeoutStr;
-    if (!context->getProperty(idleTimeout.getName(), idleTimeoutStr)
-        || !core::Property::getTimeMSFromString(idleTimeoutStr, idleTimeoutVal)) {
-      logger_->log_debug("%s attribute is invalid, so default value of %s will be used", idleTimeout.getName(),
-                         idleTimeout.getValue());
-      if (!core::Property::getTimeMSFromString(idleTimeout.getValue(), idleTimeoutVal)) {
-        assert(false);  // Couldn't parse our default value
-      }
+    if (auto idle_timeout = context->getProperty<core::TimePeriodValue>(idleTimeout)) {
+      idle_timeout_ = idle_timeout->getMilliseconds();
+    } else {
+      logger_->log_debug("%s attribute is invalid, so default value of %s will be used", idleTimeout.getName(), idleTimeout.getDefaultValue());
+      idle_timeout_ = static_cast<core::TimePeriodValue>(idleTimeout.getDefaultValue().to_string()).getMilliseconds();
```

Comment:
this `static_cast` feels "off", although at a glance I couldn't find anything in the coding guidelines that would advise against it, might even advise for it: 
> Use static_cast as the equivalent of a C-style cast that does value conversion

Reply:
you are right, I think it was due to copy paste but yeah entirely unneccesary removed it in https://github.com/apache/nifi-minifi-cpp/pull/1225/commits/78f8658fde802c73a5965dd734a544acbc37dca3 https://github.com/martinzink/nifi-minifi-cpp/commit/78f8658fde802c73a5965dd734a544acbc37dca3#diff-11bc2ef16726b961c333f05cf658ab5a88b7331722598d5d0921029e3cc47f89R174